client/packages/android/app/src/main/java/org/openmsupply/client/NativeApi.java

```diff
@@ -0,0 +1,220 @@
+        discoveryManager = (NsdManager) this.getActivity()
+                .getSystemService(NSD_SERVICE);
+
+        String debugUrl = getConfig().getString("debugUrl");
+        isDebug = debugUrl != null && !debugUrl.equals("");
+        localUrl = isDebug ? debugUrl : "https://localhost:" + discoveryConstants.PORT;
+    }
+
+    @Override
+    protected void handleOnStart() {
+        WebView webView = this.getBridge().getWebView();
+        webView.addJavascriptInterface(this, "checkCapacitor");
+        advertiseService();
+        webView.post(() -> webView.loadUrl(localUrl + "/discovery?autoconnect=true"));
+    }
+
+    @Override
+    protected void handleOnStop() {
+        stopServerDiscovery();
+    }
+
+    @PluginMethod()
+    public void goBackToDiscovery(PluginCall call) {
+        Bridge bridge = this.getBridge();
+        WebView webView = bridge.getWebView();
+        webView.post(() -> webView.loadUrl(localUrl + "/discovery?autoconnect=false"));
+    }
+
+    // Advertise local remote server on network
+    void advertiseService() {
+        NsdServiceInfo serviceInfo = new NsdServiceInfo();
+        serviceInfo.setServiceName(discoveryConstants.SERVICE_NAME);
+        serviceInfo.setServiceType(discoveryConstants.SERVICE_TYPE);
+        serviceInfo.setPort(discoveryConstants.PORT);
+        serviceInfo.setAttribute(discoveryConstants.PROTOCOL_KEY, "https");
+        serviceInfo.setAttribute(discoveryConstants.CLIENT_VERSION_KEY, "unspecified");
+        serviceInfo.setAttribute(discoveryConstants.HARDWARE_ID_KEY, discoveryConstants.hardwareId);
+
+        discoveryManager = (NsdManager) this.getActivity()
+                .getSystemService(NSD_SERVICE);
+
+        discoveryManager.registerService(serviceInfo, NsdManager.PROTOCOL_DNS_SD,
+                new NsdManager.RegistrationListener() {
+                    @Override
+                    public void onServiceRegistered(NsdServiceInfo NsdServiceInfo) {
+                    }
+
+                    @Override
+                    public void onRegistrationFailed(NsdServiceInfo serviceInfo, int errorCode) {
+                    }
+
+                    @Override
+                    public void onServiceUnregistered(NsdServiceInfo arg0) {
+                    }
+
+                    @Override
+                    public void onUnregistrationFailed(NsdServiceInfo serviceInfo, int errorCode) {
+                    }
+                });
+    }
+
+    private void stopServerDiscovery() {
+        try {
+            discoveryManager.stopServiceDiscovery(this);
+        } catch(Exception e) {
+            Logger.error("Service discovery cannot be stopped " + e );
+        }
+    }
+
+    @PluginMethod()
+    public void startServerDiscovery(PluginCall call) {
+        discoveredServers = new JSArray();
+
+        // Some navigation events may cause server discovery to still be ongoing
```

Comment:
I was getting `auth` error when navigating back to discovery, and that cause extra navigation and app crashing